feat(instrumentation/reporting): add report option (#26087)
renovatebot/renovate: docs/usage/self-hosted-configuration.md

Example URL structure: `rediss://[[username]:[password]]@localhost:6379/0`.

## reportPath

> This value describes the location where the report is written to.

Reviewer: ```suggestion
`reportPath` describes the location where the report is written to.
```

> In case [`reportType`](#reporttype) is set to `file` set this to a filepath ( e.g. `/foo/bar.json`).

Reviewer: ```suggestion
If [`reportType`](#reporttype) is set to `file`, then set `reportPath` to a filepath.
For example: `/foo/bar.json`.
```
